src/callbacks_step/amr.jl

```diff
@@ -282,14 +282,92 @@ function (amr_callback::AMRCallback)(u_ode::AbstractVector, mesh::TreeMesh,
     coarsened_original_cells = Int[]
   end
 
- # Return true if there were any cells coarsened or refined, otherwise false
- has_changed = !isempty(refined_original_cells) || !isempty(coarsened_original_cells)
- if has_changed # TODO: Taal decide, where shall we set this?
-  # don't set it to has_changed since there can be changes from earlier calls
-  mesh.unsaved_changes = true
- end
-
- return has_changed
+  # Return true if there were any cells coarsened or refined, otherwise false
+  has_changed = !isempty(refined_original_cells) || !isempty(coarsened_original_cells)
+  if has_changed # TODO: Taal decide, where shall we set this?
+    # don't set it to has_changed since there can be changes from earlier calls
+    mesh.unsaved_changes = true
+  end
+
+  if has_changed && mpi_isparallel()
```

Comment:
In general, I think - if possible - that the load balancing should be refactored into its own method and split between the generic part and the solver-specific part. I don't suggest to do this immediately - this should be the last step, once the remaining algorithm is all set up - but I didn't want to forget about it.

Reply:
What exactly do you mean by solver-specific? The element copying? Do you suggest creating a new function for all the container re-initializing? Because that's what I thought about, as we also have the same container re-initializing in `refine!` and `coarsen!` in `amr_dg2d.jl`.

Reply:
Yes, exactly like this. Everything with elements, interfaces etc is specific to the DG solver and should be refactored in an appropriate method that is called using dynamic dispatch